mapbox/src/main/java/org/odk/collect/mapbox/MapboxMapFragment.kt

```diff
@@ -295,21 +295,23 @@ class MapboxMapFragment :
             MapUtils.createPointAnnotations(requireContext(), pointAnnotationManager, markers)
 
         val featureIds = mutableListOf<Int>()
-        pointAnnotations.forEach {
-            val featureId = nextFeatureId++
-            val markerFeature = MarkerFeature(
-                requireContext(),
-                pointAnnotationManager,
-                it,
-                featureId,
-                featureClickListener,
-                featureDragEndListener,
-                MapPoint(it.point.latitude(), it.point.longitude())
-            )
+        markers.asSequence()
```

Comment:
Use [`Sequence`](https://kotlinlang.org/docs/sequences.html) here to avoid extra loops due to the `zip`.

Reply:
Hmm now this seems so complicated with that sequence and zip and it's even too late for me to wrap my head around it so I'm just going to ask: can't this whole function look like this?:
```
    override fun addMarkers(markers: List<MapFragment.MarkerDescription>): List<Int> {
        val featureIds = mutableListOf<Int>()
        markers.forEach {
            val featureId = nextFeatureId++
            val markerFeature = MarkerFeature(
                requireContext(),
                pointAnnotationManager,
                MapUtils.createPointAnnotation(pointAnnotationManager, it.point, it.isDraggable, it.iconAnchor, it.iconDrawableId, requireContext()),
                featureId,
                featureClickListener,
                featureDragEndListener,
                it.point
            )

            featureIds.add(featureId)
            features[featureId] = markerFeature
        }

        return featureIds
    }
```

Reply:
You don't find the instantiation harder to read? I also would expect your version to be slower, @grzesiek2010, though I don't know if it'd be noticeable.

Reply:
Why would my version be slower? Please notice that the original method has creating all `pointAnnotations` at the beggining:
```
        val pointAnnotations =
            MapUtils.createPointAnnotations(requireContext(), pointAnnotationManager, markers)
```
what I don't need so I removed it. Instead I create those `pointAnnotations` one by one what should take the same amount of time. Thanks to that I don't need that `asSequence` and `zip`. So I think my solution has even fewer operations and should be maybe even faster not slower.

Reply:
> Why would my version be slower?

Creating the point annotations one at a time is what caused #5177. That's why the code now creates all the point annotations at once (and up front as you say) with `PointAnnotationManager#create(options: List<PointAnnotationOptions>)`. I'm guessing that's what @lognaturel meant.

> Hmm now this seems so complicated with that sequence and zip

Agreed that it's a little awkward, especially mixing the more functional operations with a `forEach` here. I think I feel uncomfortable using `map` with side effects, but I guess I could have just split the side effect out to deal with that. I think we should leave this like this for now as the PR is in testing, but I think it'd be a cool candidate to play around with for a follow-up PR - it's got a bit of everything: needing to combine values to make other values, side effects and returning a subset values created. It's a nice opportunity to explore functional collection operations, sequences etc. Do you want to have a stab at a version you like more, and then we can discuss as a draft PR @grzesiek2010?

Reply:
Ah right i missed the fact that creating all the points at once was an improvement. Ok I can take a look at it later to see if I can come up with something better (at least to me) and then we can discuss.

Reply:
> Creating the point annotations one at a time is what caused https://github.com/getodk/collect/issues/5177.

I'll be honest, I forgot that was the root cause, it just looked like it would perform worse. 😄 If we do revisit this code, let's maybe write a comment about perf because it seems like the kind of thing that someone would restructure and accidentally cause a massive performance regression.